### PR TITLE
Adds an Android app to the add2app experimental sample

### PR DESCRIPTION
**Pull request by RedBrogdon**

### CODE REVIEW

File: experimental/add_to_app/SimpleAndroidApp/app/build.gradle [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    buildTypes {
+        release {
+            minifyEnabled false
+            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
+        }
+    }
+}
+
+dependencies {
+    implementation 'android.arch.lifecycle:common-java8:1.1.1'
```
Comment (matthew-carroll) [inline experimental/add_to_app/SimpleAndroidApp/app/build.gradle]: Just an FYI, hopefully we can remove this requirement soon when @blasten is able to add POM support in the engine.

File: experimental/add_to_app/SimpleAndroidApp/app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        <activity android:name=".MainActivity">
+            <intent-filter>
+                <action android:name="android.intent.action.MAIN"/>
+
+                <category android:name="android.intent.category.LAUNCHER"/>
+            </intent-filter>
+        </activity>
+        <activity
+            android:name="io.flutter.embedding.android.FlutterActivity"
+            android:launchMode="singleTop"
```
Comment (matthew-carroll) [inline experimental/add_to_app/SimpleAndroidApp/app/src/main/AndroidManifest.xml]: It's probably safest to leave this as-is, but I'm not sure that `singleTop` is required any more. @xster can you provide a sanity check here? Given that `FlutterEngine` is now completely independent of `FlutterActivity`, I'm thinking you should be able to have as many `FlutterActivity`s as you'd like.....

File: experimental/add_to_app/SimpleAndroidApp/app/build.gradle (hunk 1 above)
Comment (blasten) [reply experimental/add_to_app/SimpleAndroidApp/app/build.gradle]: This will be fixed by https://github.com/flutter/flutter/pull/39157

Comment (RedBrogdon) [reply experimental/add_to_app/SimpleAndroidApp/app/build.gradle]: Got it. I'll keep an eye out. These apps will need to mutate several times as y'all make progress.

File: experimental/add_to_app/SimpleAndroidApp/app/src/main/AndroidManifest.xml (hunk 2 above)
Comment (RedBrogdon) [reply experimental/add_to_app/SimpleAndroidApp/app/src/main/AndroidManifest.xml]: It should be physically impossible to launch two of these activities simultaneously in this app, so I can remove it for now. @xster, let me know what you'd prefer going forward.

**Pull request merged**